lib/zoho_invoice/base.rb

```
@@ -33,19 +30,11 @@ def self.create(client, options = {})
     def self.search(client, input_text, options = {})
       result_hash = client.get("/api/view/search/#{self.to_s.split('::').last.downcase}s", :searchtext => input_text).body
       objects_to_hydrate = result_hash['Response']["#{self.to_s.split('::').last}s"]["#{self.to_s.split('::').last}"]
-      if objects_to_hydrate.nil?
-        return []
-      else
-        objects_to_hydrate.map do |result|
-          new_hash = {}
-          result.each do |key, value|
-            new_hash[key.to_underscore.to_sym] = value if !value.is_a?(Hash) && !value.is_a?(Array)
-          end
-          self.new(client, new_hash)
-        end
-      end
+      self.process_objects(client, objects_to_hydrate)
     rescue Faraday::Error::ClientError => e
-      return []
+      if e.response[:body]
+        return  ZohoInvoice::Error::ClientError.from_response(e.response)
```

we should raise an error here instead of returning it.